test/integration/test_runner.sh

```diff
@@ -27,7 +30,8 @@ execute time go build -o "$SWARM_BINARY" ../..
 # Start the docker engine.
 execute docker --daemon --log-level=panic \
 	--storage-driver="$STORAGE_DRIVER" --exec-driver="$EXEC_DRIVER" &
-DOCKER_PID=$!
+
+trap "pkill -f docker" EXIT
```

Comment:
Only if you run it on the host. From the container it is fine. Does this ever get run on the host?

I've been trying to find another way to do this. I tried `$(cat /var/run/docker.pid)` but that isn't available immediately, so would require another retry block like below (which is maybe warranted in this case).